solutions/usaco-1181.mdx

```diff
@@ -0,0 +1,97 @@
+---
+id: usaco-1181
+source: USACO Bronze 2022 January
+title: Drought
+author: Chongtian Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_prob3_bronze_jan22.html)
+
+## Explanation
+
+We can solve this problem greedily using one linear sweep and making prefixes of cows have equal hunger as we sweep from left to right. Thus, we can break this
+problem up into cases:
+
+Suppose the cows are numbered from $1$ to $n$, and we are currently looking at cow $i$. If cow $i+1$ has a greater hunger value, then it is sufficient to decrease cow $i+1$ to the same hunger valueas cow $i$. To do this we will pick the pair of cows $i+1$ and $i+2$ and decrease their hungers until cow $i+1$ has the same hunger value as cow $i$. Of course, if cow $i+2$ has a negative hunger value, then it is impossible and we output $-1$.
+
+Now suppose cow $i$ has a greater hunger value than cow $i+1$. We obviously cannot decrease the hunger for cow $i+1$ any further since it will never be
+equal - we only can decrease the hunger for all the cows before $i+1$. To do this we select adjacent pairs of cows starting with the odd numbered cows and
+decrease their hunger values together (cows $1$ and $2$, cows $3$ and $4$, $\ldots$, cows $i-1$ and $i$). After this, all the cows before $i+1$ will be decreased to the same hunger value as cow $i+1$. However, when $i$ is odd, it is impossible to decrease it's hunger unless we also decrease the hunger value for cow $i+1$, so we output $-1$.
+
+Note that at every step we are always making sure cows from $1$ to $i+1$ have the same hunger values. 
```

Comment:
put this at the beginning lmao

Reply:
the beginning already says it

Reply:
then... why say this again

Reply:
reiterating and making what i said before in more detail now that they know what the approach is, is it necessary to remove?

Reply:
yeah delete this line